### DISCUSSION TITLE
Automatic scaling implementation question

### DISCUSSION THREAD

**Discussion by starkekr** [idaholab/moose#28496] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I am wondering how the automatic scaling feature is implemented by default. I see that there is some documentation here:
https://mooseframework.inl.gov/source/systems/NonlinearSystemBase.html. 
It seems that Jacobian scaling is implemented by default and I have a question about this. Specifically, say that our kernels set up a linear system Ax=b, does the automatic scaling set up an auxillary system, A'x=b' and solve this system (where presumably A' now has a better condition number)? Also with diagonal scaling the matrix A' is typically written in the form A'=DAE (D & E are both diagonal matrices), and b'=E^(-1)b. It seems to me that in MOOSE D is the inverse of largest absolute value of diag(A) and E is the identity. Is this correct? 

Apparently each variable also has a scaling factor thats gets added to this in some way

**Comment by starkekr** [comment]
I made a mistake in the post, b'=Db, and you recover the solution with x=E^(-1)x

**Comment by GiudGiud** [comment]
>  does the automatic scaling set up an auxillary system

We should be modifying the existing one instead of creating a new one.

> It seems to me that in MOOSE D is the inverse of largest absolute value of diag(A)

This is on a per-variable basis (well really groups of variables, they can be grouped for auto-scaling). So D isnt a single value.

> Apparently each variable also has a scaling factor thats gets added to this in some way

This is for the manual scaling system. If you use this and automatic scaling, the manual scaling factor wont do much as the automatic scaling factors will adapt to include them
